contrib/create-waku/src/cli.ts

```diff
@@ -0,0 +1,174 @@
+  let result: {
+    packageName?: string
+    shouldOverwrite?: string
+    chooseProject?: string
+  } = {}
+
+  try {
+    result = await prompts([
+      {
+        name: 'projectName',
+        type: 'text',
+        message: 'Project Name',
+        initial: defaultProjectName,
+        onState: (state: any) => (targetDir = String(state.value).trim() || defaultProjectName)
+      },
+      {
+        name: 'shouldOverwrite',
+        type: () => canSafelyOverwrite(targetDir) ? null : 'confirm',
+        message: `${targetDir} is not empty. Remove existing files and continue?`,
+      },
+      {
+        name: 'overwriteChecker',
+        type: (values: any) => {
+          if (values === false) {
+            throw new Error(red('✖') + ' Operation cancelled')
+          }
+          return null
+        }
+      },
+      {
+        name: 'packageName',
+        type: () => (isValidPackageName(targetDir) ? null : 'text'),
+        message: 'Package name',
+        initial: () => toValidPackageName(targetDir),
+        validate: (dir: string) => isValidPackageName(dir) || 'Invalid package.json name'
+      },
+      {
+        name: 'chooseProject',
+        type: 'select',
+        message: 'Choose a starter template',
+        choices: [
+          { title: 'basic-template', value: CHOICES[0] },
+          { title: 'async-template', value: CHOICES[1] },
+          { title: 'promise-template', value: CHOICES[2] },
+        ],
+      }
+    ], {
+      onCancel: () => {
+        throw new Error(red('✖') + ' Operation cancelled')
+      }
+    })
+  } catch (cancelled) {
+    if(cancelled instanceof Error) {
+      console.log(cancelled.message)
+    }
+    process.exit(1)
+  }
+
+  const { packageName, shouldOverwrite, chooseProject } = result
+
+  const root = path.join(cwd, targetDir)
+
+  if (shouldOverwrite) {
+    spawnSync('rm', ['-rf', `${root}/*`])
+  } else if (!fs.existsSync(root)) {
+    fs.mkdirSync(root)
+  }
+
+  const pkg = { name: packageName?? toValidPackageName(targetDir), version: '0.0.0' }
+
+  fs.writeFileSync(
+    path.resolve(root, 'package.json'),
+    JSON.stringify(pkg, null, 2),
+  )
+
+  console.log("Setting up project...")
+  const templateRoot = path.resolve(cwd, 'template')
+
+  const render = function render(templateName: string) {
+    const templateDir = path.resolve(templateRoot, templateName)
+
+    // Read existing package.json from the root directory
+    const existingPackageJsonPath = path.join(root, 'package.json')
+    const existingPackageJson = JSON.parse(fs.readFileSync(existingPackageJsonPath, 'utf-8'))
+
+    // Read new package.json from the template directory
+    const newPackageJsonPath = path.join(templateDir, 'package.json')
+    const newPackageJson = JSON.parse(fs.readFileSync(newPackageJsonPath, 'utf-8'))
+
+    const mergedPackageJson = {
+      ...newPackageJson,
+      ...existingPackageJson
+    }
+
+    const result = spawnSync('cp', ['-r', `${templateDir}/`, root])
```

Comment:
This probably doesn't work on Windows.
Let's use some libraries. I often  use https://www.npmjs.com/package/shelljs, but there can be some others that fit well with this use case.